frame/nfts/src/types.rs

```diff
@@ -495,3 +501,17 @@ pub struct PreSignedMint<CollectionId, ItemId, AccountId, Deadline> {
 	/// A deadline for the signature.
 	pub(super) deadline: Deadline,
 }
+
+#[derive(Clone, Eq, PartialEq, Encode, Decode, RuntimeDebug, TypeInfo)]
+pub struct PreSignedAttributes<CollectionId, ItemId, AccountId, Deadline> {
+	/// Collection's id.
+	pub(super) collection: CollectionId,
+	/// Item's id.
+	pub(super) item: ItemId,
+	/// Key-value attributes.
+	pub(super) attributes: Vec<(Vec<u8>, Vec<u8>)>,
```

Comment:
We check the length of that vec inside the code and check the length of the tuple elements as well. Since that object will be signed offchain, I would keep it as simple rust-wise as possible and don't depend on the other languages' implementation of the BoundedVec

Reply:
Yeah, I can see now where the Vec is being checked. But not where the actual tuple values are validated.

Reply:
It would also be nice to have/see the tests dedicated to length validation of all three vectors.

Reply:
```
			Self::do_set_attribute(
				signer.clone(),
				collection,
				Some(item),
				namespace.clone(),
				Self::construct_attribute_key(key)?,
				Self::construct_attribute_value(value)?,
				origin.clone(),
			)?;
```
here in `Self::construct_attribute_key(key)?` and `Self::construct_attribute_value(value)?`
it tries to put the data into the bounded vec

Reply:
Yeah, I've managed to find this also. Looks good